source/main/physics/collision/DynamicCollisions.cpp

```diff
@@ -136,6 +136,47 @@ void ResolveCollisionForces(const float penetration_depth,
     no.Forces -= gamma * forcevec;
 }
 
+/// Calculate collision forces and apply them to the collision node and the three vertex nodes of the collision triangle.
+void ResolveInterCollisionForces(const float penetration_depth,
```

Comment:
I don't like where this is going. After `calcBeamsInterTruck()` this is another fragmentation of logic between "intra-actor" and "inter-actor". But by principle, those should be perfectly the same thing - so by extension, a well designed code will only do it in one place. RoR separates it for historical reasons (rollon setting - convenient but a practical hack) and also threading in it's current design (process actors separately, then together), but IMO we should advance towards "process nodes[all actors], beams [all actors], collisions[all actors]" design.

Reply:
> this is another fragmentation of logic between "intra-actor" and "inter-actor"

I tried hard to prevent further fragmentation, but I don't think it is possible to fix submesh friction while maintaining backwards compatibility without fragmentation. I've already spent many hours to make this happen without any performance penalty and full backwards compatibility.

Reply:
I will try my best to unify those two methods once we know that it really works as intended.